membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -297,13 +350,238 @@ def test_analysis_mean(self, universe):
+        (3, 3,
+         np.array([[150., 120., 150.],
+                   [150., 120., 120.],
+                   [150., 120., 120.]])),
+        (4, 4,
+         np.array([[150., 120., 150., np.nan],
+                   [150., 120., 120., np.nan],
+                   [150., 120., 120., np.nan],
+                   [np.nan, np.nan, np.nan, np.nan]])),
+        (5, 5,
+         np.array([[150., 120., np.nan, 150., np.nan],
+                  [150., 120., np.nan, 120., np.nan],
+                  [np.nan, np.nan, np.nan, np.nan, np.nan],
+                  [150., 120., np.nan, 120., np.nan],
+                  [np.nan, np.nan, np.nan, np.nan, np.nan]]))
+
+    ])
+    def test_analysis_get_z_surface(self, universe_dummy_full, x_bin, y_bin, expected_surface):
+        mc = MembraneCurvature(universe_dummy_full,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin).run()
+        avg_surface = mc.results.average_z_surface
+        assert_almost_equal(avg_surface, expected_surface)
+
+    def test_analysis_mean(self, universe_dummy_full):
+        expected_mean = np.array([[-5.54630914e-04, - 1.50000000e+01,  8.80203593e-02],
+                                  [-2.77315457e-04, - 2.20748929e-03, - 5.01100068e-01],
+                                  [-2.77315457e-04, - 2.20748929e-03, - 1.50000000e+01]])
+        mc = MembraneCurvature(universe_dummy_full,
                                n_x_bins=3,
                                n_y_bins=3).run()
         avg_mean = mc.results.average_mean
-        assert_almost_equal(avg_mean, expected_mean)
\ No newline at end of file
+        assert_almost_equal(avg_mean, expected_mean)
+
+    @pytest.mark.parametrize('dummy_array, expected_surface', [
+        # test with negative x coordinates
+        (np.array([[0., 0., 150.], [-100., 0., 150.], [- 200., 0., 150.],
+                  [0., 100., 150.], [-100., 100., 120.], [-200., 100., 120.],
+                  [0., 200., 120.], [-100., 200., 120.], [-200., 200., 120.]]),
+         np.array([[150., 150., 120.],
+                   [150., 120., 120.],
+                   [150., 120., 120.]])),
+        # test with negative y coordinates
+        (np.array([[0., 0., 150.], [100., 0., 150.], [200., 0., 150.],
+                  [0., -100., 150.], [100., -100., 120.], [200., -100., 120.],
+                  [0., -200., 120.], [100., -200., 120.], [200., -200., 120.]]),
+            np.array([[150., 120., 150.],
+                      [150., 120., 120.],
+                      [150., 120., 120.]])),
+        # test with negative z coordinates
+        (np.array([[0., 0., -150.], [100., 0., -150.], [200., 0., -150.],
+                  [0., 100., -150.], [100., 100., 120.], [200., 100., 120.],
+                  [0., 200., 120.], [100., 200., 120.], [200., 200., 120.]]),
+            np.array([[150., 150., 120.],
+                      [150., 120., 120.],
+                      [150., 120., 120.]]))
```

Comment:
If all of these unwrap to the same coordinates, I think just the tests you already have above for the surface are enough for the x-y and you could delete the ones here. The z-unwrapping is a good thought though!

Reply:
Ok, thanks. I'll leave z coordinates.